Classes/ZedBaseHusk.uc

```diff
@@ -45,12 +45,67 @@ simulated function ToggleAuxCollision(bool newbCollision)
         super.ToggleAuxCollision(newbCollision);
 }
 
+
+// let's edit original code to avoid copy-cats in every seasonal zed class
 function SpawnTwoShots()
 {
-    if ( Controller != none )
-        super.SpawnTwoShots();
+    local vector X,Y,Z, FireStart;
+    local rotator FireRotation;
+    local KFMonsterController KFMonstControl;
+
+    // do not shoot if we are brainless, falling, dying or being moved by other husk
+    if (Controller == none || IsInState('ZombieDying') || IsInState('GettingOutOfTheWayOfShot') || Physics == PHYS_Falling)
+        return;
+
+    if (KFDoorMover(Controller.Target) != None)
+    {
+        Controller.Target.TakeDamage(22, Self, Location, vect(0, 0, 0), class'DamTypeVomit');
+        return;
+    }
+
+    GetAxes(Rotation, X, Y, Z);
+    FireStart = GetBoneCoords('Barrel').Origin;
+    
+    // TWI sets `HuskFireProjClass` both in defaultproperties and in this function...
+    // let's be smart and set it only in defaultproperties to avoid code copy-cat
+    if (!SavedFireProperties.bInitialized)
+    {
+        SavedFireProperties.AmmoClass = class'SkaarjAmmo';
+        SavedFireProperties.ProjectileClass = HuskFireProjClass;
+        SavedFireProperties.WarnTargetPct = 1;
+        SavedFireProperties.MaxRange = 65535;
+        SavedFireProperties.bTossed = false;
+        SavedFireProperties.bTrySplash = true;
+        SavedFireProperties.bLeadTarget = true;
+        SavedFireProperties.bInstantHit = false;
+        SavedFireProperties.bInitialized = true;
+    }
+
+    // Turn off extra collision before spawning vomit, otherwise spawn fails
+    ToggleAuxCollision(false);
+
+    FireRotation = Controller.AdjustAim(SavedFireProperties, FireStart, 600);
+
+    foreach DynamicActors(class'KFMonsterController', KFMonstControl)
+    {
+        // ignore zeds that the husk actually can't see, Joabyy
+        if (KFMonstControl == none || KFMonstControl == Controller || !LineOfSightTo(KFMonstControl))
+            continue;
+
+        if (PointDistToLine(KFMonstControl.Pawn.Location, vector(FireRotation), FireStart) < 75)
+        {
+            KFMonstControl.GetOutOfTheWayOfShot(vector(FireRotation),FireStart);
+        }
+    }
+
+    // added projectile owner, maybe some one will use it
+    Spawn(HuskFireProjClass, self, ,FireStart, FireRotation);
```

Comment:
Nice optimization! I bet that other event zeds can be optimized in this way too. But let leave it as a TODO item for future ;)